x/epochstorage/keeper/stake_storage.go

```diff
@@ -172,7 +172,7 @@ func (k Keeper) RemoveStakeEntryCurrent(ctx sdk.Context, chainID string, address
 	}
 
 	for i, entry := range stakeStorage.StakeEntries {
-		if entry.Address == address {
+		if entry.Operator == address || entry.Vault == address {
```

Comment:
why not &&

Reply:
because && doesn't make sense. Most of the stake entries will be with vault != operator. Also, the || lets me remove stake entries more easily because I don't have to verify that I pick vault/operator specifically